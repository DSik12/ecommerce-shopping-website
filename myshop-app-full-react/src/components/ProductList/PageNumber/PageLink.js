import React from 'react';
import { Link } from 'react-router-dom';

const PageLink = (pages) => {

    return (
        <Link to="/products">{pages.pageLinks}</Link>
    )
}

export default PageLink;