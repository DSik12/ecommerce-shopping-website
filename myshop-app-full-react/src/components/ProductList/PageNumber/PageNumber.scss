@import "../../shared/ScssBreakpoint/Breakpoint.scss";
@mixin a-styles() {
    a {
        border: 1px solid black;
        padding: 1px;
    }
}

@mixin font-style($Font_param, $display_param) {
    font-size: $Font_param;
    display: $display_param;
}

.center {
    @include font-style(10px, flex);
    margin: 1% 5% 1% 15%;
}

.pagination1 {
    @include a-styles();
}

.pagination2 {
    @include a-styles();
}

@media (min-width: $tablet-portrait) {
    .center {
        @include font-style(15px, flex);
        margin-left: 40%; /*adjust this part*/
        margin-top: 1%;
        margin-bottom: 1%;
    }
    .pagination1 .pagination2 {
        width: 100%;
        text-align: center;
    }
}
