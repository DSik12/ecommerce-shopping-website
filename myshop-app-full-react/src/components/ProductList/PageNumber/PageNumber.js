import React from 'react';
import { Link } from 'react-router-dom';
import PageLink from './PageLink';
import paths from '../productListConfig/productList.json';

const PageNumber = () => {

  let pageLinks = [];
  for (let i = 3; i < 11; i++) {
    pageLinks.push(JSON.stringify(i));
  }
  pageLinks.push("Next");
  pageLinks.push("»");
  const pages = pageLinks.map((page, index) => {
    return <PageLink pageLinks={page} key={index} />
  });


  return (
    <>
      <div className="center">
        <div className="pagination1" data-testid="pageNumberEl">
          <Link to={paths.products0to9}>«</Link>
          <Link to={paths.products0to9}>First</Link>
          <Link to={paths.products0to9}>Previous</Link>
          <Link to={paths.products0to9} id="firstPage">1</Link>
          <Link to={paths.products9to12} className="p2" id="secondPage">2</Link>
          {pages}
        </div>
      </div>
    </>

  )
}

export default PageNumber;