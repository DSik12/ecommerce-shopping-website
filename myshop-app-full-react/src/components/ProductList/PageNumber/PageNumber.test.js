import React from 'react';
import ReactDOM from 'react-dom'; 
import { render, screen } from '@testing-library/react';
import renderer from 'react-test-renderer';
import { BrowserRouter } from 'react-router-dom';
import PageNumber from './PageNumber';

describe('Pagination Component', () =>{
    it('renders the component properly', () =>{
        const divEl = document.createElement('div');
        ReactDOM.render(<BrowserRouter><PageNumber/></BrowserRouter>, divEl);
    });

    it('should have page number as text', () => {
        render(<BrowserRouter><PageNumber/></BrowserRouter>);
        const firstText = screen.getByText('First');
        expect(firstText).toBeInTheDocument();
    });

    it('should have page numbers as text', () => {
        render(<BrowserRouter><PageNumber/></BrowserRouter>);
        const previousText = screen.getByText('Previous');
        expect(previousText).toBeInTheDocument();
    });

    it('should display pagination correctly', () => {
        render(<BrowserRouter><PageNumber/></BrowserRouter>);
        const pageNumberEl = screen.getByTestId('pageNumberEl');
        expect(pageNumberEl).toHaveTextContent('«FirstPrevious12345678910Next»');
    });

    it('should have component snapshot', () => {
        const tree = renderer.create(<BrowserRouter><PageNumber /></BrowserRouter>);
        expect(tree).toMatchSnapshot();
    })
})