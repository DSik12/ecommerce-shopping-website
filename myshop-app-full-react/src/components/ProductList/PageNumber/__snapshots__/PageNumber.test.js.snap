// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pagination Component should have component snapshot 1`] = `
<div
  className="center"
>
  <div
    className="pagination1"
    data-testid="pageNumberEl"
  >
    <a
      href="/products?_start=0&_end=9"
      onClick={[Function]}
    >
      «
    </a>
    <a
      href="/products?_start=0&_end=9"
      onClick={[Function]}
    >
      First
    </a>
    <a
      href="/products?_start=0&_end=9"
      onClick={[Function]}
    >
      Previous
    </a>
    <a
      href="/products?_start=0&_end=9"
      id="firstPage"
      onClick={[Function]}
    >
      1
    </a>
    <a
      className="p2"
      href="/products?_start=9&_end=12"
      id="secondPage"
      onClick={[Function]}
    >
      2
    </a>
    <a
      href="/products"
      onClick={[Function]}
    >
      3
    </a>
    <a
      href="/products"
      onClick={[Function]}
    >
      4
    </a>
    <a
      href="/products"
      onClick={[Function]}
    >
      5
    </a>
    <a
      href="/products"
      onClick={[Function]}
    >
      6
    </a>
    <a
      href="/products"
      onClick={[Function]}
    >
      7
    </a>
    <a
      href="/products"
      onClick={[Function]}
    >
      8
    </a>
    <a
      href="/products"
      onClick={[Function]}
    >
      9
    </a>
    <a
      href="/products"
      onClick={[Function]}
    >
      10
    </a>
    <a
      href="/products"
      onClick={[Function]}
    >
      Next
    </a>
    <a
      href="/products"
      onClick={[Function]}
    >
      »
    </a>
  </div>
</div>
`;
